### PR TITLE
upd765: Implement SRB status register, including RDDATA bit.  

### PR DESCRIPTION
**Pull request by shattered**
Replace magic numbers. (nw)

Add dvk_mx floppy format.  Combined, this lets MXONPC run in at586 driver.

### CODE REVIEW

File: src/devices/imagedev/floppy.cpp [hunk 1]
```diff
@@ -784,6 +784,25 @@ attotime floppy_image_device::get_next_index_time(std::vector<uint32_t> &buf, in
 	return base + attotime::from_nsec((uint64_t(next_position)*2000/floppy_ratio_1+1)/2);
 }
 
+int floppy_image_device::get_rddata(const attotime &when)
```
[the remark this answers is not in the document]
Comment (shattered) [reply src/devices/imagedev/floppy.cpp]: rddata in this register is not the raw signal:

http://www.textfiles.com/bitsavers/pdf/intel/_dataSheets/290410-001_82077SL_Floppy_Disk_Controller_May91.pdf

"The two TOGGLE bits [WRDATA TOGGLE, RDDATA TOGGLE] do not read back the state of their respective pins directly.
Instead, the pins drive a Flip/Flop which produces a wider and more reliably read pulse."

And I've verified that "MXONPC" software works fine with this implementation -- it samples the rddata bit and does FM decoding.

**Comment by galibert** [comment]
If it's working fine, it's by sheer luck.  You're returning the polarity
direction (A or B), rddata is supposed to return polarity direction
*changes*.

  OG.

**Comment by shattered** [comment]
I'm doing that because output of the flip-flop is essentially the polarity direction:

http://ohlandl.ipv7.net/floppy/DS011332.pdf

"Every positive edge transition of the RDATA disk interface output causes this bit to change states."

**Comment by galibert** [comment]
Interesting that it's a flip-flop, that explains why it didn't entirely fuck up.  It's still wrong though, it will fail badly in the presence of weak bits or degaussed regions...

**Comment by shattered** [comment]
I'll try to rework this then.

**Pull request closed without merging**